build/clean-up-cluster.sh

```diff
@@ -4,6 +4,21 @@
 
 set -e
 
+function delete_all_and_wait() {
+    RESOURCE=$1
+    NAMESPACE=$2
+    COUNT=$3
+    SKIP_DELETE=$4
+    if [ "$SKIP_DELETE" != "true" ] ; then
+        oc delete $RESOURCE -n $NAMESPACE --all --ignore-not-found
+    fi
+    while [ $(oc get $RESOURCE -n $NAMESPACE --no-headers | wc -l | tr -d '[:space:]') -gt $COUNT ]
+    do
+        oc get $RESOURCE -n $NAMESPACE
+        sleep 2
+    done
```

Comment:
This could loop forever, if a controller is unexpectedly recreating a resource and the COUNT is 0. It seems like the point is to ensure the count gets to the desired number when it's non-zero, can that just be an additional condition on this loop?

Reply:
My main point is, this could loop forever, there is no timeout. Should I be worried about that?

Reply:
oh, yeah, I will add a timeout

Reply:
done